File: src/SWEA/python/SWE_4014_활주로건설.py
# N * N 의 활주로 건설
def check(row):
    global N, X
    cnt = 1
    for i in range(1, N):
        if row[i] == row[i-1]:	# 같은 높이라면
            cnt += 1
        elif row[i] - row[i-1] == 1 and cnt >= X:   # 높이 1 높아지면
            cnt = 1
        elif row[i-1] - row[i] == 1 and cnt >= 0:   # 높이 1 낮아지면
            cnt = -X + 1
        else:   # 높이 2 이상 차이나면
            return 0
    if cnt >= 0:
        return 1
    return 0

for tc in range(1, int(input()) + 1):
    N, X = map(int, input().split())
    graph = [list(map(int, input().split())) for _ in range(N)]
    answer = 0

    for i in range(N):
        answer += check(graph[i])
        
        temp = []
        for j in range(N):
            temp.append(graph[j][i])
        
        answer += check(temp)

    print(f"#{tc} {answer}")

File: src/SSAFY/day221011/SWE_4014_활주로건설.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class SWE_4014_활주로건설 {
    static int N, X;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int TC = Integer.parseInt(br.readLine());

        for (int tc = 1; tc <= TC; tc++) {
            int answer = 0;
            st = new StringTokenizer(br.readLine());

            N = Integer.parseInt(st.nextToken());
            X = Integer.parseInt(st.nextToken());

            int[][] graph = new int[N][N];

            for (int i = 0; i < N; i++) {
                st = new StringTokenizer(br.readLine());

                for (int j = 0; j < N; j++) {
                    graph[i][j] = Integer.parseInt(st.nextToken());
                }
            }

            for (int i = 0; i < N; i++) {
                answer += check(graph[i]);

                int[] temp = new int[N];
                for (int j = 0; j < N; j++) {
                    temp[j] = graph[j][i];
                }
                answer += check(temp);
            }

            System.out.printf("#%d %d\n", tc, answer);

        }
    }

    static int check(int[] temp){
        int cnt = 1;

        for (int i = 1; i < N; i++) {
            if(temp[i] == temp[i - 1]) cnt += 1;
            else if(temp[i] - 1 == temp[i - 1] && cnt >= X) cnt = 1;
            else if(temp[i] + 1 == temp[i - 1] && cnt >= 0) cnt = -X + 1;
            else return 0;
        }
        if(cnt >= 0) return 1;
        return 0;
    }
}
